app/src/main/java/org/schabi/newpipe/fragments/detail/VideoDetailFragment.java

```diff
@@ -445,6 +447,14 @@ public void onClick(final View v) {
                 break;
             case R.id.detail_controls_playlist_append:
                 if (getFM() != null && currentInfo != null) {
+                    final Fragment fragment = getParentFragmentManager().
+                            findFragmentById(R.id.fragment_holder);
+
+                    // commit previous pending changes to database
+                    if (fragment instanceof LocalPlaylistFragment) {
+                        ((LocalPlaylistFragment) fragment).commitChanges();
+                    }
```

Comment:
What about the cases when the local playlist fragment is a main page tab?

Reply:
This case can also be handled by these lines.
I tested this case on my device. It works fine.

Reply:
Mmmh, I don't understand why those lines would handle it. If the fragment is a main page tab, then it is not the parent of the VideoDetailFragment anymore

Reply:
I have added a supplementary video for this case. Did I understand you correctly?

Reply:
Fixed in 2de828289f91ef28ad0960fc784f102fc52bde5b